week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -18,36 +18,113 @@ This means the order the songs are played in will be random, but each song will
 
  */
 
-
 class MusicPlayer {
-    // Add your code here
-
+  // Add your code here
+  constructor() {
+    this.playlist = [];
+    this.playing = false;
+    this.trackIndex = 0;
+  }
+  add(track, artist) {
+    let song = {};
+    song.name = track;
+    song.artist = artist;
+    this.playlist.push(song);
+  }
+
+  play(choice) {
+    if (this.playlist.length == 0) {
+      console.log("You have no songs in your playlist!");
+    } else {
+      if (choice) {
+        console.log(
+          `Currently playing:${this.playlist[choice].name} by ${this.playlist[choice].artist}`
+        );
```

Comment:
Might be a better design approach logically in this case if you assign the `choice` value to the `trackIndex` variable to be the new index position. In that way, play function would be in charge of changing and storing new track index if there is.
But it needs some code refactoring. Have to consider pros and cons, I might be wrong.